### PR TITLE
feat: Allow base kernel to be selected

### PR DESCRIPTION
**Pull request by mseeger**
Apart from allowing the base GP kernel to be selected, this contains a number of doc fixes.

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py [hunk 1]
```diff
@@ -226,7 +223,7 @@ def __init__(
                 )
 
     @property
-    def ARD(self) -> float:
+    def ARD(self) -> bool:
```
Comment (wesk) [inline syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py:226]: Typehint was wrong?

Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py:226]: Yes

Comment (wesk) [reply syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py:226]: Yes, I see it was being set to a boolean, resolving

File: syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py [hunk 2]
```diff
@@ -133,27 +133,24 @@ def _inverse_bandwidths(self):
...
 
+        X1_scaled = anp.multiply(X1, inverse_bandwidths)
+        X1_squared_norm = anp.sum(anp.square(X1_scaled), axis=1)
         if X2 is X1:
-            X1_scaled = anp.multiply(X1, inverse_bandwidths)
             D = -2.0 * anp.dot(X1_scaled, anp.transpose(X1_scaled))
-            X1_squared_norm = anp.sum(anp.square(X1_scaled), axis=1)
-            D = D + anp.reshape(X1_squared_norm, (1, -1))
-            D = D + anp.reshape(X1_squared_norm, (-1, 1))
+            X2_squared_norm = X1_squared_norm
```
Comment (wesk) [inline syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py:147]: This line and the removal of the old line 148, are the only things that still confuse me in the PR. 

Do we really not need the old line 148 any more? It's handled by new lines 147, 152, 153?

Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py:147]: 148 is now 153.

**Review by wesk — APPROVED** [review]

**Pull request merged**